### DISCUSSION TITLE
ProtonMail Integration

### DISCUSSION THREAD

**Discussion by jumpingbrowntrout** [simple-login/app#1061] [Q&A]
Firstly congratulations @nguyenkims and team for joining Proton family and your efforts in making a great email privacy product.
It's been great to be apart of that journey. 

I would like to know if there is anywhere where primary plans are being shared about integration between the two services and where the best place for me to share my ideas (I have a few :))

**Comment by JasonMoriarty** [comment]
I'm just very curious about the progress and want to know when the specifics will be available to everyone.

**Reply by nguyenkims** [reply]
The integration will happen over many phases. In the first phase, the integration will be quite "light" with the main goals of 1) allowing Proton users to quickly have SimpleLogin account to start hiding their Proton address and 2)help Proton users discover email aliases concept.

This phase has already been started and we hope to be able to release it in the next coming months! 

Future phases will allow a deeper integration of SimpleLogin into Proton and any ideas are indeed welcome!

**Reply by JasonMoriarty** [reply]
Can't wait to use it

**Reply by xncelt** [reply]
### SPAM folder offer.  Is that all?

Other than the SPAM folder offer, I don't see any other integration of SimpleLogin (which I love) with Protonmail (which I also love).  Am I missing something?

My number one suggestion would be to have a button available when you create your SL alias giving you option of creating a reverse -alias and adding it to your Protonmail contacts with the correct syntax.

1. currently I have to open two MS Windows side-by-side (1) SL dashboard (2) my protonmail account.
2. I create the alias in SL.
3. I create a reverse alias by clicking SL "Contacts".
4. I change focus to the Protonmail window.
5. I copy and paste the revese alias to to protonmail contacts which usually rejects it.
6. I open notepad and remove everything before the < and the trailing >
7. example:  "My-name Alias | my-name+mailbox at aleeas.com" <my-name_mailbox_at_aleeas_com_rndegok@simplelogin.co>

**Reply by nguyenkims** [reply]
@xncelt please stay tuned, the development of a deeper integration is already starting and we hope to release it soon :)

**Comment by jumpingbrowntrout** [comment]
There's a few key things I'd like to see which I think would be killer

1) Ability to fire off an email to a new contact using their email (not the reverse alias) and have PM/SL create an alias on the fly or use an existing one if previous created. Creating reverse alias's require a few manual steps so be nice to automate this.

2) Do away with mail being received from the reverse alias. With SL API integration we should be able to show the true sender within PM. This will neaten up the inbox and make searches display the true sender.

3) Would be nice if we could block a sender in PM which blocks the sender in SL.

To send mail with the SL service, you could make a SL place-holder address which can be selected from the 'from' field. You could also set this as your default address if you wish to use SL as the primary address. 

I really don't think anything else is needed. All the admin for alias's and SL service it self should be contained to SL. We just need the mail composing and blocking features in PM.

**Reply by JasonMoriarty** [reply]
I recently use StartMail.(oc main on Gmail + ProtonMail) What they do to create an alias is really worth learning. Just put a button on the right hand of the Sender. Then ask when will the alias expire.

**Comment by d4rklynk** [comment]
I would like to know about pricing plans integration if possible. Does Simplelogin will be cheaper for paid plan protonmail users ? Will it be integrated into paid plan of protonmails ?

**Comment by xncelt** [comment]
Son Nguyen Kim,

What has been happening with your work to integrate SimpleLogin "deeper" with ProtonMail since I last checked with you in February?  I check your YouTube channel often.  There has not been an update for one year.  Is this a full-time job for you?  How much priority is the staff at ProtonMail giving to your efforts?

**Reply by nguyenkims** [reply]
@xncelt hi, unfortunately I can't give an ETA on when the integration will be finished. As it often happens with softwares, integration in a big project can take time and more complexity can be revealed during the process.

To answer the other question, this it's a full time for me :). Currently my time is shared between SimpleLogin and the new product Proton Pass (which relies on SimpleLogin for the email alias part).

**Comment by xncelt** [comment]


**Reply by markcellus** [reply]
@xncelt not sure if you're aware but your emailed response has your phone number in it. If personal details like this aren't something you want exposed on the internet, you may want to remove it before responding to Github through your email account.

**Comment by xncelt** [comment]
Is ProtonPass the Protonmail and Simplelogin integration we have been waiting for?  It looks like what Protonmail calls "Hide-my-email" is Simplelogin integrated and renamed.  Is that accurate?

**Reply by JasonMoriarty** [reply]
Not, that is another thing.

**Reply by xncelt** [reply]
I found the answer to my question under the FAQ's on the following website https://proton.me/pass/alias.  

"Does Proton Pass use SimpleLogin by Proton?

Proton Pass uses [SimpleLogin by Proton(new window)](https://simplelogin.io/), which is an email alias service that provides advanced email forwarding. You can also manage your aliases in SimpleLogin itself.

I understand that Proton Pass was just released today so informtion about how to import all of my SimpleLogin aliases into Proton Pass may not be available yet.  Son Nguyen Kim, can you provide this information.

**Reply by nguyenkims** [reply]
Hi we're working on making SL aliases available in Proton Pass too, please stay tuned.

> Is ProtonPass the Protonmail and Simplelogin integration we have been waiting for

There's going to be more integration between Proton Mail and SL that will make using SL easier.

**Reply by springcomp** [reply]
@nguyenkims Does ProtonPass support self-hosted SimpleLogin instances for managing aliases ?

**Reply by nguyenkims** [reply]
@springcomp Currently not but please feel free to submit the idea on Proton Pass feature request forum https://protonmail.uservoice.com/forums/953584-proton-pass so other people can upvote the idea :)

**Comment by xncelt** [comment]
Great, thanks for the reply. Looks good so far. My wife keeps saying to me that this alias thing confuses her but, I tell her that with ProtonPass is working to solve that.

Roger Newhall
***@***.***
215-510-2255

**Comment by nathanstill** [comment]
The best workflow I can think of would be to leave the contact list alone (mine is getting messy/confusing with all the reverse aliases) and just make simple login domains available from the "from" list with an empty box before the at sign to allow me to type the from address.  The alias (if not existing) and the reverse alias (if not existing for the chosen recipients) would then be generated on the fly and more or less transparent to the Proton user.
